### PR TITLE
Remove 32 bit code from DriverInterface & update project build

### PR DESCRIPTION
**Pull request by nealsid**
settings to build for macOS 13.

### CODE REVIEW

File: src/MacMSRDriver/DriverInterface.c [hunk 1]
```diff
@@ -1,153 +1,80 @@
...
-	}
-	else {
-		kernResult = IOConnectMethodScalarIScalarO(connect, kOpenDriver, 0, 0);
-	}    
-#endif
-    
+        kern_return_t   kernResult;
+
+        kernResult = IOConnectCallScalarMethod(connect, kOpenDriver, NULL, 0, NULL, NULL);
 	return kernResult;
```
Comment (opcm) [inline src/MacMSRDriver/DriverInterface.c:13]: should it return IOConnectCallScalarMethod(connect, kOpenDriver, NULL, 0, NULL, NULL); directly to simplify code?

File: src/MacMSRDriver/DriverInterface.c [hunk 2]
```diff
@@ -1,153 +1,80 @@
...
-#if !defined(__LP64__)
-	}
-	else {
-		kernResult = IOConnectMethodScalarIScalarO(connect, kCloseDriver, 0, 0);
-	}    
-#endif
-	
-    return kernResult;
+
+        kernResult = IOConnectCallScalarMethod(connect, kCloseDriver, NULL, 0, NULL, NULL);
```
Comment (opcm) [inline src/MacMSRDriver/DriverInterface.c:21]: similar

File: src/MacMSRDriver/DriverInterface.c [hunk 3]
```diff
@@ -1,153 +1,80 @@
...
-#if !defined(__LP64__)
-	}
-	else {
-		kernResult = IOConnectMethodStructIStructO(connect, kReadMSR, *idata_size, odata_size, idata, odata);
-	}
-#endif
-    return kernResult;
+
+        kernResult = IOConnectCallStructMethod(connect,	kReadMSR, idata, *idata_size, odata, odata_size);
+        return kernResult;
```
Comment (opcm) [inline src/MacMSRDriver/DriverInterface.c:31]: and here

File: src/MacMSRDriver/DriverInterface.c [hunk 4]
```diff
@@ -1,153 +1,80 @@
...
-	}
-	else {
-		kernResult = IOConnectMethodStructIStructO(connect, kWriteMSR, *idata_size, NULL, data, NULL);
-	}
-#endif
-    
-    return kernResult;
+
+        kernResult = IOConnectCallStructMethod(connect,	kWriteMSR, (void*)data, *idata_size, NULL, NULL);
+        return kernResult;
```
Comment (opcm) [inline src/MacMSRDriver/DriverInterface.c:39]: and here

File: src/MacMSRDriver/DriverInterface.c [hunk 5]
```diff
@@ -1,153 +1,80 @@
...
-	}
-	else {
-		kernResult = IOConnectMethodStructIStructO(connect, kBuildTopology, 0, data_size, NULL, data);
-	}
-#endif
-    
-    return kernResult;
+
+        kernResult = IOConnectCallStructMethod(connect,	kBuildTopology, NULL, 0, data, data_size);
+        return kernResult;
```
Comment (opcm) [inline src/MacMSRDriver/DriverInterface.c:47]: and here

**Review by opcm — CHANGES_REQUESTED** [review]
thanks

File: src/MacMSRDriver/DriverInterface.c (hunk 1 above)
Comment (nealsid) [reply src/MacMSRDriver/DriverInterface.c]: Done.

File: src/MacMSRDriver/DriverInterface.c (hunk 2 above)
Comment (nealsid) [reply src/MacMSRDriver/DriverInterface.c]: Done.

File: src/MacMSRDriver/DriverInterface.c (hunk 3 above)
Comment (nealsid) [reply src/MacMSRDriver/DriverInterface.c]: Done.

File: src/MacMSRDriver/DriverInterface.c (hunk 4 above)
Comment (nealsid) [reply src/MacMSRDriver/DriverInterface.c]: Done.

File: src/MacMSRDriver/DriverInterface.c (hunk 5 above)
Comment (nealsid) [reply src/MacMSRDriver/DriverInterface.c]: Done.

**Comment by nealsid** [comment]
Good call, thanks!

**Review by opcm — APPROVED** [review]

**Pull request merged**